qiskit_experiments/framework/common_operations.py

```diff
@@ -0,0 +1,47 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""A collection of common operation callback in execution chain."""
+
+
+from qiskit.providers import Backend
+from qiskit.test.mock import FakeBackend
+
+from .base_experiment import BaseExperiment
+
+
+def apply_delay_validation(experiment: BaseExperiment, backend: Backend):
```

Comment:
Does this need to be in framework? It really feels like all the content of this function is something that should be handled correctly by terra transpiler/scheduler.

Reply:
Half of this code is now automatically handled by transpiler since backends recently started to report the timing constraints information. However, we still need to set scheduling options, i.e. transpiler options of `scheduling_method`. This can be a default transpiler options of corresponding experiments, however, a fake backend doesn't require scheduling since it implicitly calls simulator and no timing constraints there. But we can still set scheduling option for simulator backend at the expense of transpiler overhead (this is heavy compute pass since it recreates DAG circuits).

Reply:
I tried not to change currently implemented code, but I can also update that code in this PR. There are two options
- Keep this pre-transpile. Remove the code to set `acquire_alignment`.
- Entirely remove this code and set `"scheduling_method": "asap"` to default transpiler options. This will induce non-necessary overhead in the simulator execution.